Can't compile on CentOS 8 Stream

Running `configure` ends with:

```
checking for openssl/pem.h... yes
checking openssl/rsa.h usability... yes
checking openssl/rsa.h presence... yes
checking for openssl/rsa.h... yes
checking openssl/sha.h usability... yes
checking openssl/sha.h presence... yes
checking for openssl/sha.h... yes
checking openssl/ssl.h usability... yes
checking openssl/ssl.h presence... yes
checking for openssl/ssl.h... yes
checking for library containing ERR_peek_error... none required
checking for library containing SSL_library_init... no
configure: error: Cannot build shared opendkim
                                              against static openssl libraries.
                                              Configure with --disable-shared
                                              to get this working or obtain a
                                              shared libssl library for
                                              opendkim to use.
Command exited with non-zero status 1
```

Running `configure  --disable-shared` ends with:
```
checking for openssl/pem.h... yes
checking openssl/rsa.h usability... yes
checking openssl/rsa.h presence... yes
checking for openssl/rsa.h... yes
checking openssl/sha.h usability... yes
checking openssl/sha.h presence... yes
checking for openssl/sha.h... yes
checking openssl/ssl.h usability... yes
checking openssl/ssl.h presence... yes
checking for openssl/ssl.h... yes
checking for library containing ERR_peek_error... none required
checking for library containing SSL_library_init... no
checking for library containing SSL_library_init... no
configure: error: libssl not found
Command exited with non-zero status 1
```

openssl-devel *is* installed.

--- a/configure.ac
+++ b/configure.ac
@@ -888,9 +888,12 @@ then
 	               AC_MSG_ERROR([libcrypto not found]))
 
 
+	# SSL_library_init() was removed in OpenSSL 1.1.0 (it became a no-op
+	# macro).  Use SSL_CTX_new(), which is a real function in all supported
+	# OpenSSL versions (1.0.x, 1.1.x, 3.x).
 	AC_LINK_IFELSE(
 		       [AC_LANG_PROGRAM([[#include <openssl/ssl.h>]],
-					[[SSL_library_init();]])],
+					[[SSL_CTX_free(NULL);]])],
 					[od_have_ossl="yes";],
 					[od_have_ossl="no";])
 	if test x"$od_have_ossl" = x"no"
@@ -906,7 +909,7 @@ then
 		fi
 
 		LIBCRYPTO_LIBS="$LIBCRYPTO_LIBS -ldl"
-		AC_SEARCH_LIBS([SSL_library_init], [ssl], ,
+		AC_SEARCH_LIBS([SSL_CTX_new], [ssl], ,
 		               AC_MSG_ERROR([libssl not found]), [-ldl])
 	fi
 

